packages/driver/src/cypress/mocha.ts

```diff
@@ -37,6 +37,97 @@ delete (window as any).Mocha
 
 export const SKIPPED_DUE_TO_BROWSER_MESSAGE = ' (skipped due to browser)'
 
+// NOTE: 'calculateTestStatus' is marked as an individual function to make functionality easier to test.
+export function calculateTestStatus (test: Mocha.Test, strategy?: 'detect-flake-and-pass-on-threshold' | 'detect-flake-but-always-fail', options?: any) {
```

Comment:
What do you think about using our own type that extends `Mocha.Test` here? That way it will have all the properties/methods we expect and don't need to use so many `// @ts-expect-error` comments.

Reply:
I like the idea, but there will be at least one place we need to coerce the type, likely when we call the function in the prototype. I can work on a commit for this

Reply:
Took a stab at this in 1af7ed3303. Not perfect but should be in a better spot